magma/circuit.py

```diff
@@ -151,13 +147,10 @@ def _has_definition(cls, port=None):
     if port is None:
         interface = getattr(cls, "interface", None)
         if not interface:
-            return None
+            return False
         return any(_has_definition(cls, p) for p in interface.ports.values())
-    if isinstance(port, Tuple) or isinstance(port, Array):
-        return any(_has_definition(cls, elem) for elem in port)
-    if not port.is_output():
-        return port.value() is not None
-    return False
+    flat = port.flatten()
+    return any([not f.is_output() and f.value() is not None for f in flat])
```

Comment:
Don't need the extra list comp before any I think `[...]`.  Ideally `flatten()` returns a generator object so we don't have to fully expand this in a lot of cases.

Reply:
Done. Will make `flatten` return a generator in a future optimization